src/applications/apps.js

```diff
@@ -50,12 +50,18 @@ export function registerApplication(appName, arg1, arg2) {
   if (typeof activeWhen !== 'function')
     throw new Error(`The activeWhen argument must be a function`);
 
-  apps.push({
-    name: appName,
-    loadImpl,
-    activeWhen,
-    status: NOT_LOADED,
-  });
+  const app = {
+      name: appName,
+      loadImpl,
+      activeWhen,
+      status: NOT_LOADED,
+  };
+
+  if (customProps) {
+      app.customProps = customProps;
```

Comment:
Let's validate customProps type and throw an error if it is something unexpected.

As mentioned above, I think we should decide on whether to accept just objects or accept anything. I think I lean towards just objects, since that's what the word "props" usually connotes and an object is what is passed as props to lifecycle functions, anyway.

Your thoughts?

Example of what I mean:
```js
if (typeof customProps !== 'object') {
  throw new Error('customProps must be an object');
}
```

Reply:
I don't mind restricting it on just objects. As long as we don't restrict the type of the objects properties as well, since I am currently passing functions as properties.

Reply:
👍 agreed